include/rocksdb/db.h

```diff
@@ -163,6 +163,12 @@ class DB {
                      const std::vector<ColumnFamilyDescriptor>& column_families,
                      std::vector<ColumnFamilyHandle*>* handles, DB** dbptr);
 
+  // Close the DB by releasing resources, closing files etc. This should be
+  // called before calling the desctructor so that the caller can get back a
+  // status in case there are any errors. Regardless of the return status, the
+  // DB must be freed
+  virtual Status Close() = 0;
```

Comment:
I suggest we have a default which returns Status::OK(). Otherwise, any user with a class inherited from class DB will get a build error, which is not a good experience after they upgrade to a newer version.